internal/pubkey/encode.go

```diff
@@ -22,11 +22,12 @@ import (
 
 // EncodeByProtocol selects the correct encoding method, then encodes the public key with it.
 func EncodeByProtocol(in []byte, protocol string) (encoded, encodingType string, err error) {
+	encodingType = encoding.TypeHex0XPrefix
```

Comment:
I would suggest moving this inside the case statement for each case. It then is clearer for someone looking at this in the future that `encodingType` can change based on protocol. Although its more code, it is clearer. Always best to be clear.